### DISCUSSION TITLE
Alltalk Google jupyter notebook notebook

### DISCUSSION THREAD

**Discussion by johnbenac** [erew123/alltalk_tts#139] [Q&A] [answered]
Can someone make a colab notebook for alltalk? That would be sweet. Then I could run this in colab, and make the voices, while I had my little 1070 RAM making images. 

Also, I'd like to train models with alltalk on colab, because it takes all night to do it on my machine. 

I tried this notebook:

https://colab.research.google.com/github/Vali-98/xtts-finetune-webui/blob/main/xtts_finetune.ipynb

But it had an error when training. 

Alltalk on Colab might be a gamechanger and get a lot more sillytavern users (and other users, like ooba) because you wouldent have to have a GPU to use it!!

And unlike SD, or even some LLMs, you dont have to configure alltalk hardly at all once you get it going. 

Does this already exist? It shouldent be too hard to do. 

Maybe I'll do it if no one else has. 

It may be easier than setting up on a local machine!

For reference, here is an official Coque colab notebook: 

https://colab.research.google.com/github/camenduru/coqui-XTTS-colab/blob/main/coqui_XTTS_v2_colab.ipynb

But that doesnt have training on it.

**Comment by johnbenac** [comment]
hmm. I tried creating my own here: https://colab.research.google.com/drive/1GtvZUeO8BtgAbGRZE5-r8PYsD_0M8RQq?usp=sharing

But it gets hung up when running docker.

**Comment by erew123** [comment, marked as answer]
Hi @johnbenac 

I've had a brief pop at this in the past and may return to it at some point. The actual issue, which also affects updating docker at the moment, is the requirements that need to be installed by PIP. There are some minor conflicts, which can be resolved during a normal installation on a desktop computer at time of installation, but when running inside a colab or docker build, the virtual machine requires a restart to handle the conflict & installing a different version of the same package e.g.

Lets say the requirement for `numpty` is version 1.8 and you install version 1.8 as part of the requirements file, but a bit further down the list, something else in the list demands that numpty version 1.1 is installed. All of a sudden the colab hangs saying something to the jist of "you need to restart to update numpty to version 1.1".

On a desktop computer, this would just go through and you could force and upgrade to version 1.8 again. On a colab you cant. 

I've spent a little time looking at it, but its complicated to deal with, as you changing any of the other requirements can cause knock on effects. Its something I've been slowly working towards trying to resolve by tightening up other requirements, however there are code changes involved in that. So its a bit by bit process.

If anyone wants to take a pop at it, they are welcome to. Somewhere I think I have the "as far as I got" script for google colab, but cant immediately find it on my machine.

Thanks

**Reply by johnbenac** [reply]
hey, thanks for explaining that. Hopefully, if others are looking, they wont waste the time that I did just trying to do it thoughtlessly
